src/hypertable_insert.c

```diff
@@ -471,16 +471,16 @@ ts_hypertable_insert_path_create(PlannerInfo *root, ModifyTablePath *mtpath)
 								"constraints"),
 						 errhint("Use column names to infer indexes instead.")));
 
+			subpath = ts_chunk_dispatch_path_create(root, mtpath, rti, i);
```

Comment:
I am OK with this change, but it is not an obvious improvement IMO. Usually, you build a tree of paths recursively, so it is logical that the parent creates the child, which in turn creates its own children and so on recursively. This change breaks that pattern where the hypertable_insert path creates the child of the child. This breaks the normal pattern.